src/client/java/io/wispforest/limelight/impl/ui/ResultsContainerComponent.java

```diff
@@ -58,11 +63,17 @@ private void rebuildContents() {
             clearChildren();
 
             for (var entry : results) {
-                var result = new ResultEntryComponent(screen, entry);
+                boolean isExpanded = entry instanceof ExpandableResultEntry && expanded.getOrDefault(entry.entryId(), false);
+                var result = new ResultEntryComponent(screen, entry, false, isExpanded);
 
                 if (screen.firstResult == null) screen.firstResult = result;
 
                 child(result);
+
+                if (isExpanded) {
+                    for (ResultEntry e : ((ExpandableResultEntry)entry).getChildren())
+                        child(new ResultEntryComponent(screen, e, true, false));
+                }
```

Comment:
I assume this just doesn't do anything if a child entry of an expandable entry is also expandable?